src/query_bed.py

```diff
@@ -3,15 +3,27 @@
 import argparse  # we use this module for option parsing. See main for details.
 
 import sys
+from turtle import up
 from bed import (
     read_bed_file, print_line, BedLine
 )
-
+from bounds import (
+    lower_bound, upper_bound
+)
 
 def extract_region(features: list[BedLine],
-                   start: int, end: int) -> list[BedLine]:
+                   start: int, end: int,) -> list[BedLine]:
     """Extract region chrom[start:end] and write it to outfile."""
-    return []  # FIXME: We want the actual region, not an empty list!
+    lst = []
+    for bedline in features:
+        lst.append(bedline.chrom_start)
```

Comment:
This is a bit unfortunate.

You copy the start positions for all the features on the chromosome, which takes time $O(n)$ and then you use the lower and upper bound to search in $O(\log n)$. True, you don't extract the chromosome for each query--I'm happy to see that--but it isn't necessary to extract the start points at all. You could just adapt the search functions so they compare against the start positions of the features. Then you would save $O(n)$ and only need the $O(\log n)$ queries.